pkg/server/api/enroll.go

```diff
@@ -103,6 +105,33 @@ func clientHostnameMatches(hostname string, r *http.Request) bool {
 	return cert.VerifyHostname(hostname) == nil
 }
 
+func clientSpiffeIdMatches(expected string, r *http.Request) (bool, error) {
+	conn := r.TLS
+	if len(conn.VerifiedChains) == 0 {
+		return false, nil
+	}
+
+	cert := conn.VerifiedChains[0][0]
+
+	// Get the SPIFFE ID from the presented certificate
+	actualId, err := x509svid.IDFromCert(cert)
+	if err != nil {
+		return false, nil
+	}
+
+	// Form a good spiffe ID from the configuration string
```

Comment:
good call to use the spiffe library to ensure we're dealing with valid SPIFFE IDs!